pkg/printer/printer.go

```diff
@@ -1,9 +1,24 @@
 package printer
 
 import (
+	"fmt"
+
 	"github.com/doitintl/kube-no-trouble/pkg/judge"
 )
 
+var printers = map[string]func() Printer{
+	"json": newJSONPrinter,
+	"text": newTextPrinter,
+}
+
 type Printer interface {
 	Print([]judge.Result) error
 }
+
+func ParsePrinter(choice string) (func() Printer, error) {
+	printer, exists := printers[choice]
+	if !exists {
+		return nil, fmt.Errorf("failed to validate argument output")
```

Comment:
We should not mention argument in this module
```suggestion
		return nil, fmt.Errorf("unknown printer type %s", choice)
```